app/controllers/subscriptions.js

```diff
@@ -135,6 +194,11 @@ export default Ember.Controller.extend(Ember.Evented, {
 
   // each action below is an event in a channel
   update_store: function(data, success) {
+    if (this.applyCustomHandler(data)) {
+      run(success);
+      return;
+    }
+
```

Comment:
I've also added this.
Basically, if there's a custom handler for this type, use that instead of the default behaviour